networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -1044,26 +1076,28 @@ def johnson(G, weight='weight', new_weight=None):
     if not nx.is_weighted(G, weight=weight):
         raise nx.NetworkXError('Graph is not weighted.')
 
-    new_node = nx.utils.generate_unique_node()
-    G.add_weighted_edges_from((new_node, node, 0) for node in G.nodes())
+    dist = {}
+    pred = {}
+    for node in G:
+        dist[node] = 0
+        pred[node] = None
```

Comment:
Simpler as a dictionary comprehension: `dist = {v: 0 for v in G}` and `pred = {v: None for v in G}`.